static/app/views/starfish/queries/useSpanSamples.tsx

```diff
@@ -52,7 +52,8 @@ export const useSpanSamples = (options: Options) => {
     groupId ? {group: groupId} : undefined,
     {transactionName, 'transaction.method': transactionMethod},
     [`p95(${SPAN_SELF_TIME})`],
-    'sidebar-span-metrics'
+    'sidebar-span-metrics',
+    Boolean(groupId && transactionName && transactionMethod)
```

Comment:
Maybe instead of manually passing this in, we decide if it is enabled in the `useSpanMetricsSeries` function?
That way its more dynamic, don't need to manually pass it in everytime.
something like
```
const isEnabled = Object.values(queryFilters).every(value => Boolean(value))
```

Reply:
👍 Moved the enabled check inside `useSpanMetricsSeries`
Didn't update `useSpanMetrics` since it looks like there are scenarios where transactionName and transactionMethod filters are intentionally not defined

Reply:
Perfect, lgtm!
 🤔 do you mind linking a scenario where transactionName and transactionMethod are intentionally not defined? I'm just curious why they would be intentionally undefined, vs just not passed in at all.

Reply:
yeah its this component here:
https://github.com/getsentry/sentry/blob/master/static/app/views/starfish/views/spanSummaryPage/index.tsx#L56

The span summary page can be filtered to an endpoint + method but it can also be unfiltered.
technically the name and method aren't being passed in, the query filter is being defaulted to {}, but I didn't like the way the type was defined because it opens up the possibility to pass in an undefined name or method:
```
type SpanSummaryQueryFilters = {
  'transaction.method'?: string;
  transactionName?: string;
};
```
Let me update this type and function in a follow up pr